src/main/java/seedu/uninurse/model/PersistentUninurseBook.java

```diff
@@ -0,0 +1,89 @@
+package seedu.uninurse.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import seedu.uninurse.commons.core.Config;
+import seedu.uninurse.logic.commands.CommandResult;
+
+/**
+ * Versions of UninurseBookSnapshot after every command.
+ */
+public class PersistentUninurseBook extends UninurseBook {
+
+    private final List<UninurseBookSnapshot> uninurseBookVersions;
+    private int currentVersion;
+
+    /**
+     * Creates an UninurseBookSnapshot using the Persons in the {@code toBeCopied}
+     */
+    public PersistentUninurseBook(ReadOnlyUninurseBook toBeCopied) {
+        super(toBeCopied);
+        this.uninurseBookVersions = new ArrayList<UninurseBookSnapshot>();
+        this.uninurseBookVersions.add(new UninurseBookSnapshot(toBeCopied));
+        this.currentVersion = 0;
+        this.handleChange();
+    }
+
+    private void handleChange() {
+        setPersons(this.uninurseBookVersions.get(currentVersion).getPersonList());
+    }
+
+    /**
+     * Returns whether you can revert to an earlier version of UninurseBook.
+     */
+    public boolean canUndo() {
+        return currentVersion > 0;
+    }
+
+    /**
+     * Returns whether you can revert to a later version of UninurseBook.
+     */
+    public boolean canRedo() {
+        return currentVersion + 1 < uninurseBookVersions.size();
+    }
+
+    /**
+     * Reverts to an earlier version of UninurseBook.
+     */
+    public void undo() {
+        if (canUndo()) {
+            currentVersion -= 1;
+            handleChange();
+        }
+    }
+
+    /**
+     * Reverts to a later version of UninurseBook.
+     */
+    public void redo() {
+        if (canRedo()) {
+            currentVersion += 1;
+            handleChange();
+        }
+    }
```

Comment:
Minor issue but I realize that you tend to use `var -= 1` and `var += 1` instead of `var--` and  `var++`. Maybe too much python? xD

Reply:
I think it's just my general coding style to use `var++` and `var--` only in `for` loops. I'll change this for consistency.